jx/ui/Box.jxml

```diff
@@ -64,14 +78,16 @@
 					var child = attr.children[k],
 						delta_child = delta.children[k];
 
-					if (!child[flow_dim]) delta_child[flow_dim] = Math.max(size, MIN_SIZE);
-					if (!child[full_dim]) delta_child[full_dim] = full_size;
+					if (!(flow_dim in child)) delta_child[flow_dim] = Math.max(size, MIN_SIZE);
+					if (isPercentage(child[flow_dim])) delta_child[flow_dim] = my_size * parseFloat(child[flow_dim]) / 100.0;
+					if (!(full_dim in child)) delta_child[full_dim] = full_size;
 
 					delta_child[full_axis] = padding;
 					delta_child[flow_axis] = position;
-					position += gap + (child[flow_dim] || delta_child[flow_dim]);
+					position += gap + (delta_child[flow_dim] || child[flow_dim]);
 				}
 			}
+			 function isPercentage(p) { return /^\d*%$/.test(p) }
```

Comment:
this code doesn't match percentages with decimal points. try this regex instead.

```js
/^\d+(\.\d+)?$/
```
another example of a place where it would be nice if we could have a utility method with tests...